gen/templates/generic/data_source.go

```diff
@@ -152,18 +158,37 @@ func (d *{{camelCase .Name}}DataSource) Read(ctx context.Context, req datasource
 		return
 	}
 
-	tflog.Debug(ctx, fmt.Sprintf("%s: Beginning Read", config.Id.String()))
 
-	res, err := d.client.Get("{{if .GetRestEndpoint}}{{.GetRestEndpoint}}{{else}}{{.RestEndpoint}}{{end}}" + config.Id.ValueString())
+	{{ if not .RemoveId}}
+	tflog.Debug(ctx, fmt.Sprintf("%s: Beginning Read", config.Id.String()))
+	{{- end}}
+	{{if .RemoveId}}
```

Comment:
this should become an "else" block of the previous "if" statement.

Reply:
Ah these were technically separate and unrelated. Will change with the above as well